docs/reference/inference/service-cohere.asciidoc

```diff
@@ -0,0 +1,183 @@
+[[infer-service-cohere]]
+=== Cohere {infer} `service`
+
+Creates an {infer} endpoint to perform an {infer} task with the `cohere` service.
+
+
+[discrete]
+[[infer-service-cohere-api-request]]
+==== {api-request-title}
+
+`PUT /_inference/<task_type>/<inference_id>`
+
+[discrete]
+[[infer-service-cohere-api-path-params]]
+==== {api-path-parms-title}
+
+`<inference_id>`::
+(Required, string)
+include::inference-shared.asciidoc[tag=inference-id]
+
+`<task_type>`::
+(Required, string)
+include::inference-shared.asciidoc[tag=task-type]
++
+--
+Available task types:
+* `completion`,
+* `rerank`,
+* `text_embedding`.
+--
+
+[discrete]
+[[infer-service-cohere-api-request-body]]
+==== {api-request-body-title}
+
+`service`::
+(Required, string)
+The type of service supported for the specified task type. In this case, 
+`cohere`.
+
+`service_settings`::
```

Comment:
I think it'd probably make sense to add 3 collapsable sections for cohere because there are various service settings for each task type. We forgot to add the text embedding ones so we could do that on a follow up PR if you'd like.

Reply:
Yeah sure:

`completion`
`model_id`: `(Optional, string) The name of the model to use for the inference task.` (We can include a link to the models just link the blurb about rerank if we want.
`rate_limit ` (same as what's in the top section, it'll be the same for all 3 task types)


`rerank`
Same as `completion`

`text_embedding`
`similarity`: Optional, Similarity measure, one of `cosine`, `dot_product`, `l2_norm`. Defaults based on the `embedding_type` (float -> dot product, int8/byte -> cosine).
`embedding_type` same as before
`model_id` Same as top section
`rate_limit` Same as top section